sdk/src/commitment_config.rs

```diff
@@ -46,11 +46,28 @@ impl CommitmentConfig {
 
 #[derive(Serialize, Deserialize, Clone, Copy, Debug, PartialEq, Eq, Hash)]
 #[serde(rename_all = "camelCase")]
+/// An attribute of a slot. It describes how finalized a block is at some point in time. For example, a slot
+/// is said to be at the max level immediately after the cluster recognizes the block at that slot as
+/// finalized. When querying the ledger state, use lower levels of commitment to report progress and higher
+/// levels to ensure state changes will not be rolled back.
 pub enum CommitmentLevel {
+    /// The highest slot confirmed by the cluster as having reached `MAX_LOCKOUT_HISTORY` confirmations. The
+    /// validator will not recognize forks below this slot.
     Max,
+
+    /// The highest slot of the heaviest fork. Ledger state at this slot is not derived from a finalized
```

Comment:
I think it would be nice to add that this slot has also been voted on by the connected node. But maybe this wording was intentional to apply to non-voting nodes?

Reply:
Exactly. While the current implementation might only update the cache after voting, I don't think that should be required. It's simply the slot the validator *would* vote on, if it had to.  There's a number of reasons why it might not vote or delay its vote.